src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Globalization/UseOrdinalStringComparison.Fixer.cs

```diff
@@ -0,0 +1,139 @@
+                // "a != b" => "!string.Equals(a, b, StringComparison.Ordinal)"
+                context.RegisterCodeFix(new MyCodeAction(SystemRuntimeAnalyzersResources.StringComparisonShouldBeOrdinalOrOrdinalIgnoreCase,
+                                async ct => await FixEquals(context.Document, syntaxFactoryService, root, node, context.CancellationToken).ConfigureAwait(false)),
+                        diagnostic);
+            }
+        }
+
+        protected abstract bool IsInArgumentContext(SyntaxNode node);
+        protected abstract Task<Document> FixArgument(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode argument);
+
+        protected abstract bool IsInIdentifierNameContext(SyntaxNode node);
+        protected abstract Task<Document> FixIdentifierName(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode identifier, CancellationToken cancellationToken);
+
+        protected abstract bool IsInEqualsContext(SyntaxNode node);
+        protected abstract Task<Document> FixEquals(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode node, CancellationToken cancellationToken);
+
+        internal SyntaxNode CreateEqualsExpression(SyntaxGenerator syntaxFactoryService, SemanticModel model, SyntaxNode operand1, SyntaxNode operand2, bool isEquals)
+        {
+            var stringType = model.Compilation.GetSpecialType(SpecialType.System_String);
+            var memberAccess = syntaxFactoryService.MemberAccessExpression(
+                        syntaxFactoryService.TypeExpression(stringType),
+                        syntaxFactoryService.IdentifierName(UseOrdinalStringComparisonAnalyzer.EqualsMethodName));
+            var ordinal = CreateOrdinalMemberAccess(syntaxFactoryService, model);
+            var invocation = syntaxFactoryService.InvocationExpression(
+                memberAccess,
+                operand1,
+                operand2.WithoutTrailingTrivia(),
+                ordinal)
+                .WithAdditionalAnnotations(Formatter.Annotation);
+            if (!isEquals)
+            {
+                invocation = syntaxFactoryService.LogicalNotExpression(invocation);
+            }
+
+            invocation = invocation.WithTrailingTrivia(operand2.GetTrailingTrivia());
+
+            return invocation;
+        }
+
+        internal SyntaxNode CreateOrdinalMemberAccess(SyntaxGenerator syntaxFactoryService, SemanticModel model)
+        {
+            var stringComparisonType = WellKnownTypes.StringComparison(model.Compilation);
+            return syntaxFactoryService.MemberAccessExpression(
+                syntaxFactoryService.TypeExpression(stringComparisonType),
+                syntaxFactoryService.IdentifierName(UseOrdinalStringComparisonAnalyzer.OrdinalText));
+        }
+
+        protected bool CanAddStringComparison(IMethodSymbol methodSymbol)
```

Comment:
Shouldn't we also check that the method is declared on well known System.String type?